docs/source/howto/workflows.rst

```diff
@@ -4,14 +4,217 @@
+
+.. note::
+
+    For more information on the ``define()`` method and the process spec, see the :ref:`corresponding section in the topics <topics:processes:usage:defining>`.
+
+.. code-block:: python
+
+    def multiply(self):
+        """Multiply two integers."""
+        self.ctx.multiple = multiply(self.inputs.x, self.inputs.y)
+
+The ``multiply`` method is the first step in the outline of the ``MultiplyAddWorkChain`` work chain.
+This step simply involves running the calculation function ``multiply``, on the ``x`` and ``y`` **inputs** of the work chain.
+To store the result of this function and use it in the next step of the outline, it is added to the *context* of the work chain using ``self.ctx``.
+
+.. code-block:: python
+
+    def add(self):
+        """Add two numbers with the ArithmeticAddCalculation process."""
+
+        inputs = {'x': self.ctx.multiple, 'y': self.inputs.z, 'code': self.inputs.code}
+        calcjob_node = self.submit(ArithmeticAddCalculation, **inputs)
+
+        return ToContext({'addition': calcjob_node})
+
+The ``add`` method is the second step in the outline of the work chain.
+As this step uses the ``ArithmeticAddCalculation`` calculation job, we start by setting up the inputs for this ``CalcJob`` in a dictionary.
+Next, when submitting this calculation job to the daemon, it is important to use the submit method from the work chain instance via ``self.submit()``.
+Since the result of the addition is only available once the calculation job is finished, the ``submit()`` method returns the ``CalcJobNode`` of the *future* ``ArithmeticAddCalculation`` process.
+To tell the work chain to wait for this process to finish before continuing the workflow, we return the ``ToContext`` class, where we have passed a dictionary to specify that the future calculation job node should be assigned to the ``'addition'`` context key.
+
+.. note::
+    Instead of passing a dictionary, you can also initialize a ``ToContext`` instance by passing the future process as a keyword argument, e.g. ``ToContext(addition=calcjob_node)``.
+    More information on the ``ToContext`` class can be found in :ref:`the topics section on submitting sub processes<topics:workflows:usage:workchains:submitting_sub_processes>`.
+
+.. code-block:: python
+
+    def validate_result(self):
+
+        result = self.ctx['addition'].outputs.sum
+
+        if result.value < 0:
+            return self.exit_codes.ERROR_NEGATIVE_NUMBER
+
+Once the ``ArithmeticAddCalculation`` calculation job is finished, the next step in the work chain is to validate the result, i.e. verify that the result is not a negative number.
+After the ``addition`` node has been extracted from the context, we take the ``sum`` node from the ``ArithmeticAddCalculation`` outputs and store it in the ``result`` variable.
+In case the value of this ``Int`` node is negative, the ``ERROR_NEGATIVE_NUMBER`` exit code - defined in the ``define()`` method - is returned.
```

Comment:
We should add here that when an exit code is returned from any outline step, the work chain will be terminated. No other steps will be executed